### ISSUE TITLE
dotCMS V5.0.1 open redirect

### ISSUE THREAD

**Issue by howchen** [dotCMS/core#15286]
## Current Behavior
dotCMS V5.0.1 allows remote attackers to redirect users to arbitrary web sites and conduct phishing attacks via a malformed URL
e.g.
```
http://localhost:8999/html/common/forward_js.jsp?FORWARD_URL=http://www.baidu.com
http://localhost:8999/html/portlet/ext/common/page_preview_popup.jsp?hostname=google.com
```

![redirect](https://user-images.githubusercontent.com/14739143/45534977-70c91480-b82f-11e8-9ae4-2a789c9b05f0.png)

## Possible Solution
 
limit the redirect target

**Comment by wezell** [comment]
Thank you for this report.  This issue will be resolved in 5.0.2 with the code from this issue:
#15274

**Comment by valentijnscholten** [comment]
Please note this issue also affects the 3.x and 4.x versions.

Looking at the commit in #15724 it looks like the chosen solution is to mandate authentication for these urls. Wouldn't that leave logged in users still vulnerable?
